### PR TITLE
Configure new plugin registry and broker

### LINKED ISSUE #12908: Merge all che_plugin.yml fields into meta.yaml
### Description

Currently the informations to load plugins are splitted in two distinct files: meta.yml and che_plugin.yml. See for example che-theia [meta.yml](https://github.com/eclipse/che-plugin-registry/blob/master/plugins/org.eclipse.che.editor.theia/next/meta.yaml) and [che-plugin.yml](https://github.com/eclipse/che-theia/blob/master/che-plugins/che-editor-theia/etc/che-plugin.yaml).

We have experienced that managing 2 distinct files is problematic: 
- some information are duplicated
- some kind of plugins can't have a che_plugin.yaml (i.e. vs code extensions)
- clients may need access to the informations included in the che_plugin.yml (i.e. the user dashboard and the memory limits of a plugin)

The proposed solution is to extend the meta.yaml model to include all the fields of a che_plugin.yml and to remove the need for che_plugin.yml file at all.

### PR DESCRIPTION
**Pull request by garagatyi**
In WIP because required broker version is not released yet.

### What does this PR do?
This PR bring changes introduced by PR https://github.com/eclipse/che-plugin-broker/pull/55.
In particular it merges che-plugin.yaml and meta.yaml of Che 7 plugins.
See detailed description of cahnges in broker in the above mentioned PR.
It also removes not-needed anymore classes and rename `editorCommands` field to `commands`.

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/12908

### CODE REVIEW

**Comment by garagatyi** [comment]
@ibuziuk Because of the renaming of `editorCommands` field to `commands` we won't be able to adopt new broker without upgrading to a new upstream version.

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/ChePlugin.java [hunk 1]
```diff
@@ -11,36 +11,126 @@
...
+
+  public ChePlugin publisher(String publisher) {
+    this.publisher = publisher;
     return this;
   }
 
-  public List<EditorCompatibility> getEditors() {
-    if (editors == null) {
-      editors = new ArrayList<>();
+  public String getVersion() {
```
Comment (amisevsk) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/ChePlugin.java:79]: Method order could be a more consistent here, e.g have the chained method, then `set` then `get`.

**Review by amisevsk — APPROVED** [review]
LGTM

**Review by davidfestal — APPROVED** [review]

**Comment by garagatyi** [comment]
ci-test

**Comment by amisevsk** [comment]
Needs plugin broker release `v0.17` before merging.

**Comment by monaka** [comment]
@amisevsk  This can be closed since #13311 was opened ?

**Comment by amisevsk** [comment]
@monaka Yes, thank you.

**Pull request closed without merging**